src/datadog/grok/matchers/date.rs

```diff
@@ -239,125 +251,119 @@ pub fn apply_date_filter(value: &Value, filter: &DateFilter) -> Result<Value, Gr
                 value.pop(); // drop Z
                 value.push_str("+0000");
             }
-            match &filter.regex_with_tz {
-                Some(re) => {
-                    let tz = re
-                        .captures(&value)
-                        .ok_or_else(|| {
+            if let Some(tz) = filter
```

Comment:
There are two real changes here in the `apply_date_filter` function:
 - instead of  the removed `regex_with_tz`, which used to be optional, we check the regex for the named <tz> group.
 - we call `replace_sec_fraction_with_dot(filter, &mut value);` to replace fraction character with dot, since `strftime` format supports only dot-leading format `%.f` for a fraction of a second.
 
The rest is formatting.

The rest is reformatting.